tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs

```diff
@@ -21,5 +29,151 @@ public static void Constructors_VerifyGuardClauses()
+            CatalogueItemType itemType,
+            OrderItemValidator orderItemValidator)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
+                new OrderItemRecipientModel { DeliveryDate = order.CommencementDate.Value.AddDays(-1) }).ToList();
+
+            var result = orderItemValidator.Validate(order, new CreateOrderItemModel { ServiceRecipients = serviceRecipients }, itemType);
+
+            result.FailedValidations.Should().NotBeEmpty();
+
+            foreach ((_, ValidationResult validationResult) in result.FailedValidations)
+            {
+                var errorDetails = validationResult.Errors.SingleOrDefault();
+                errorDetails.Should().NotBeNull();
+                errorDetails.Field.Should().Be("DeliveryDate");
+                errorDetails.Id.Should().Be("DeliveryDateOutsideDeliveryWindow");
+            }
+
+            result.Success.Should().BeFalse();
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Validate_DeliveryDateAfterMaxDeliveryDateOffsetInDays_SuccessIsFalse(
+            Order order,
+            CreateOrderItemModel model,
+            CatalogueItemType itemType,
+            ValidationSettings validationSettings)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+
+            OrderItemValidator orderItemValidator = new OrderItemValidator(validationSettings);
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
+                new OrderItemRecipientModel { DeliveryDate = order.CommencementDate.Value.AddDays(validationSettings.MaxDeliveryDateOffsetInDays + 1) }).ToList();
+
+            var result = orderItemValidator.Validate(order, new CreateOrderItemModel { ServiceRecipients = serviceRecipients }, itemType);
+
+            result.FailedValidations.Should().NotBeEmpty();
+
+            foreach ((_, ValidationResult validationResult) in result.FailedValidations)
+            {
+                var errorDetails = validationResult.Errors.SingleOrDefault();
+                errorDetails.Should().NotBeNull();
+                errorDetails.Field.Should().Be("DeliveryDate");
+                errorDetails.Id.Should().Be("DeliveryDateOutsideDeliveryWindow");
+            }
+
+            result.Success.Should().BeFalse();
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Validate_DeliveryDateWithinMaxDeliveryDateOffsetInDays_SuccessIsTrue(
+            Order order,
+            CreateOrderItemModel model,
+            CatalogueItemType itemType,
+            ValidationSettings validationSettings)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+
+            itemType.Should().NotBeEquivalentTo(CatalogueItemType.AssociatedService);
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
```

Comment:
As above, as `r` isn't used we can use a discard here.

Reply:
removed `r`